File: samplejava/src/main/java/com/example/chattutorialjava/ChannelViewModelsFactory.java
package com.example.chattutorialjava;

import com.getstream.sdk.chat.viewmodel.ChannelHeaderViewModel;
import com.getstream.sdk.chat.viewmodel.MessageInputViewModel;
import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

@SuppressWarnings("unchecked")
public class ChannelViewModelsFactory extends ViewModelProvider.NewInstanceFactory {

    private final String cid;

    public ChannelViewModelsFactory(@NonNull String cid) {
        this.cid = cid;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass == MessageListViewModel.class) {
            return (T) new MessageListViewModel(cid);
        } else if (modelClass == ChannelHeaderViewModel.class) {
            return (T) new ChannelHeaderViewModel(cid);
        } else if (modelClass == MessageInputViewModel.class) {
            return (T) new MessageInputViewModel(cid);
        } else {
            return super.create(modelClass);
        }
    }
}

File: app/src/main/java/com/example/chattutorial/ChannelViewModelsFactory.kt
package com.example.chattutorial

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.getstream.sdk.chat.viewmodel.ChannelHeaderViewModel
import com.getstream.sdk.chat.viewmodel.MessageInputViewModel
import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel

class ChannelViewModelsFactory(private val cid: String) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return when (modelClass) {
            MessageListViewModel::class.java -> {
                MessageListViewModel(cid) as T
            }
            ChannelHeaderViewModel::class.java -> {
                ChannelHeaderViewModel(cid) as T
            }
            MessageInputViewModel::class.java -> {
                MessageInputViewModel(cid) as T
            }
            else -> {
                super.create(modelClass)
            }
        }
    }
}
